contracts/starknet/src/entry/structs.cairo

```diff
@@ -61,6 +61,19 @@ struct Currency {
     ethereum_address: felt,  // optional
 }
 
+struct FuturesCurrency {
+    id: felt,
+    decimals: felt,
+    is_abstract_currency: felt,  // True (1) if not a specific token but abstract, e.g. USD or ETH as a whole
+    starknet_address: felt,  // optional, e.g. can have synthetics for non-bridged assets
+    ethereum_address: felt,  // optional
+}
+
+struct GenericEntity {
+    id: felt,
+    decimals: felt,
+    type_of_data: felt,
```

Comment:
what's the purpose of this struct ?

Reply:
This structure is to provide a general structure for non-currency related informations

Reply:
Alright makes sense, although I think it could just be integrated to the `Currency` struct w/ an additional `type` field. What do you think?

Reply:
I think it could be possible for simpler code. However, in the future, we will still have to make a completely different entity to handle elements that do not have decimals (NFTs?).